gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TripEdgeArrivalDepartureTimeValidatorTest.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.notice.MissingTripEdgeStopTimeNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+
+@RunWith(JUnit4.class)
+public class TripEdgeArrivalDepartureTimeValidatorTest {
+  private final GtfsTrip trip1 =
+      new GtfsTrip.Builder().setCsvRowNumber(1).setTripId("trip1").build();
+  private final GtfsTrip trip2 =
+      new GtfsTrip.Builder().setCsvRowNumber(2).setTripId("trip2").build();
+
+  private final GtfsStopTime startStopTimeHasBothTrip1 =
+      new GtfsStopTime.Builder()
+          .setCsvRowNumber(18)
+          .setTripId("trip1")
+          .setStopId("1001")
+          .setStopSequence(1)
+          .setArrivalTime(GtfsTime.fromString("12:02:34"))
+          .setDepartureTime(GtfsTime.fromString("12:03:34"))
+          .build();
+  private final GtfsStopTime startStopTimeHasArrivalTrip1 =
```

Comment:
perhaps it would be more useful for understanding this test if we named such stop as startStopNoDepartureTrip1

Reply:
ok has been updated to reflect this for all variables

Reply:
Thanks